### DISCUSSION TITLE
Delete triggering ModelPropertyChanged for BusinessListBase but not BusinessBindingListBase in Blazor

### DISCUSSION THREAD

**Discussion by michaelcsikos** [MarimerLLC/csla#2951] [Questions]
I have two collections, one inherits from `BusinessListBase`, and the other from `BusinessBindingListBase`. Each collection is wrapped in a `ViewModel<T>`. The `PropertyChangedEventHandler` from the root edit page is passed down to all the children:
```cs
// Root edit page
private async void PropertyChanged(object? sender, PropertyChangedEventArgs e)
{
    await InvokeAsync(() => StateHasChanged());
}

// In each child collection
[Parameter]
public PropertyChangedEventHandler PropertyChanged { get; set; }

protected override void OnInitialized()
{
    VM.ModelPropertyChanged += PropertyChanged;
}
```

In the collection `ViewModel`, child items are deleted like this:
```cs
 public void Remove(WidgetVM context)
 {
     if (Model == null)
         return;

     Model    .Remove(context.Model);
     WidgetVMs.Remove(context);
 }
```

For the `BusinessListBase` deleting a child item triggers `PropertyChanged`, it bubbles up to the root edit page and the **Save** button becomes enabled, but it's not bubbling up for the `BusinessBindingListBase`. Ordinarily I would just use `BusinessListBase` to solve it, but we have some classes which are used in WinForms as well.

Am I approaching this correctly, i.e. making a VM to wrap each collection, and a VM to wrap each child item, and passing `PropertyChangedEventHandler` down?

**Comment by rockfordlhotka** [comment]
I created issue #2954 for this, as I suppose there should be feature parity - or at least it is worth considering.

Regarding wrapping each domain object with a viewmodel, yes, probably. I'm becoming increasingly frustrated with MVVM outside the XAML world, as it seems to add complexity. I haven't found something _better_ yet, but it feels like there should be a better option.

**Reply by michaelcsikos** [reply]
@rockfordlhotka I share your frustration. I feel like I'm repeating myself a lot with wide and deep object graphs.

**Reply by rockfordlhotka** [reply]
Exactly!
